plugins/dell-kestrel/fu-dell-kestrel-hid-device.c

```diff
@@ -258,14 +261,25 @@ fu_dell_kestrel_hid_device_write_firmware_pages(FuDellKestrelHidDevice *self,
 				fu_progress_step_done(progress);
 				continue;
 			}
-			g_propagate_error(error, g_steal_pointer(&error_local));
+			g_propagate_prefixed_error(error,
+						   g_steal_pointer(&error_local),
+						   "%s failed to write page: ",
+						   fu_device_get_name(FU_DEVICE(self)));
 			return FALSE;
 		}
 
-		/* device needs time to process incoming pages */
-		if (j == 0) {
-			g_debug("wait %u ms before the next page", first_page_delay);
-			fu_device_sleep(FU_DEVICE(self), first_page_delay);
+		/* rmm needs extra time to accept incoming pages */
+		if ((j == 0) && (dev_type == FU_DELL_KESTREL_EC_DEV_TYPE_RMM)) {
+			FuDevice *dev = FU_DEVICE(self);
+			FwupdVersionFormat fmt = fu_device_get_version_format(dev);
+			const gchar *version = fu_device_get_version(dev);
+
+			if (fu_version_compare(version, "1.8.6", fmt) < 0) {
```

Comment:
I'd honestly just use the fu_device_get_version_format() and fu_device_get_version() here -- you don;t need the values again.